File: C/aula9.c
#include <stdio.h>
#define GREEN "\x1b[32m"
#define YELLOW "\x1b[3;34;43m"
#define RESET "\x1b[0m"
#define INVERT "\x1b[7m"

void saudar(const char[], char[]);

void main(){
    const char SAUDACAO[] = "Oi, ";
    //const char SAUDACAO[] = {79,105,44,32,0};
    //const char SAUDACAO[] = {'O', 'i',',',' ','\0'};
    
    char nome[15]= "";      // = ""; evita comportamento indefinido (lixo ou falha)

    printf(GREEN "LITERAIS, CONSTANTES, VARIAVEIS\n" RESET);

    printf("\n1 - LITERAL CLASSICO HELLOWORLD:\n");
    printf(INVERT "Oi, mundo!\n" RESET);

    printf("\n2 - VALOR PADRAO DA VARIAVEL ANTES DA INICIALIZACAO:\n");
    saudar(SAUDACAO, nome);

    printf(YELLOW "\nATRIBUICOES EM DESIGN-TIME\n" RESET);   // TEMPO DE PROJETO
    printf("\n3 - INICIALIZACAO:\n");
    sprintf(nome, "Ana");      // não poderia inicializar dessa forma: nome = "Ana";
    saudar(SAUDACAO, nome);

    printf("\n3 - NOVA ATRIBUICAO:\n");
    sprintf(nome, "Bela");      // não poderia atribuir dessa forma: nome = "Bela";
    saudar(SAUDACAO, nome);

}

void saudar(const char S[], char n[]){
    printf("nome: '%s'\n", n);
    printf("%s%s!\n", S, n);
}

File: Python/aula9.py
import sys, random
INVERT = "\x1b[1;7m"
RESET = "\x1b[0m"
GREEN = "\x1b[1;32m"
YELLOW = "\x1b[1;34;43m"

K = random.randint(1,9)
n1 = None
DATA = [1, 3, 5, 7, 9]
READ = iter(DATA)

print(f"{GREEN}LITERAL, CONSTANTE e VARIAVEL{RESET}")

print(f"\nLiteral numerico magico:\t{INVERT}1089{RESET}")

print("\nValor da variavel antes da inicializacao: ")
print(f"n1 = {n1}")

print(f"\n{YELLOW}Atribuicoes em design-time{RESET}")
n1 = 3
print("Inicializacao direta: ")
print(f"n1 = {n1}")

n1 = next(READ);    n1 = next(READ);    n1 = next(READ)
print("\nNova atribuicao simulando read/data: ")
print(f"n1 = {n1}")

print(f"\n{YELLOW}Atribuicao em invocation-time{RESET}")    # python argv[0] argv[1]
n1 = int(sys.argv[1]) if len(sys.argv) > 1 and sys.argv[1].isdigit() else 0
print("Entrada por linha de comando: ")
print(f"n1 = {n1}")

print(f"\n{YELLOW}Atribuicao em runtime{RESET}")      
print("Entrada por interacao: ")
x = input("Digite um numero (0-9): ")
n1 = int(x) if x.isdigit() else 0                   
print(f"n1 = {n1}")

print(f"\n{GREEN}OPERACOES ARITMETICAS{RESET}")      
print(f"n1 + K = {n1 + K}")
print(f"n1 - K = {n1 - K}")
print(f"n1 * K = {n1 * K}")
print(f"n1 / K = {n1 / K}")
print(f"n1 // K = {n1 // K}")
print(f"n1 % K = {n1 % K}")
print(f"n1 ** K = {n1 ** K}")

print(f"\n{GREEN}D E S A F I O{RESET}")      
x = input("Adivinhe o valor da constante K (1-9): ")
palpite = int(x) if x.isdigit() else 0                   
print(f"Acertou! " if palpite == K else "Quase! ", end=""); print(f"Era {K}.");


